### ISSUE TITLE
Nested configs aren't applied when `--path` is provided.

### ISSUE THREAD

**Issue by bclymer** [realm/SwiftLint#1825]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
`0.22.0`
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
`Homebrew`

Project structure
```
./Root.swift
./.swiftlint.yml
./Tests/Test.swift
./Tests/.swiftlint.yml
```

`./Root.swift`
```swift
public class Root {
    private let test = Int("1")!
    private let test2 = "" as! NSString
}
```

`./.swiftlint.yml`
```yml
disabled_rules:
 - force_unwrapping
```

`./Tests/Test.swift`
```swift
public class Test {
    private let test = Int("1")!
    private let test2 = "" as! NSString
}
```

`./Tests/.swiftlint.yml`
```yml
disabled_rules:
 - force_cast
```

Running `swiftlint` from `./` prints

```
Loading configuration from '.swiftlint.yml'
Linting Swift files in current working directory
Linting 'Root.swift' (1/2)
Linting 'Test.swift' (2/2)
/Users/brianclymer/Desktop/SwiftLintTest/Root.swift:3:28: error: Force Cast Violation: Force casts should be avoided. (force_cast)
Done linting! Found 1 violation, 1 serious in 2 files.
```

Running `swiftlint --path Tests/Test.swift` from `./` prints

```
Loading configuration from '.swiftlint.yml'
Linting Swift files at path Tests/Test.swift
Linting 'Test.swift' (1/1)
/Users/brianclymer/Desktop/SwiftLintTest/Tests/Test.swift:3:28: error: Force Cast Violation: Force casts should be avoided. (force_cast)
Done linting! Found 1 violation, 1 serious in 1 file.
```

The bug is that `Tests.swift` shouldn't report any errors, as the nested configuration should be loaded, but isn't. This only happens when `--path` is provided.

**Comment by SDGGiesbrecht** [comment]
@bclymer,

There was a lot of discussion about this about a month ago. The discussion was scattered throughout several closed issues and pull requests, so I am not surprised you did not find it in your search. I am not aware that the maintainers ever published a decision on what `--path` even *should* do.

I recommend reading [this in‐depth comment](https://github.com/realm/SwiftLint/pull/1745#issuecomment-320802347). Then post here making a case for your understanding of the purpose of `--path`. (It looks like probably an instance of situation A from that post?) Then wait for a response from a maintainer (of which I am not).
